fiftyone/server/files/paths.py

```
@@ -0,0 +1,131 @@
+    # character rather than a path separator).
+    safe_path = path.replace("\\", "/")
+
+    try:
+        normalized = fos.realpath(safe_path)
+        base = fos.realpath(allowed_base)
+    except (OSError, ValueError) as e:
+        raise FileOperationError(
+            code=FileOperationErrorCode.PATH_INVALID,
+            message=f"Invalid path: {e}",
+            details={"path": path},
+        )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)